### PR TITLE
Validation of batch types

### PR DESCRIPTION
**Pull request by Poltergeist**
This is adding the support of type validation in the `Transaction`s 

I choose to use `Symbols` as values over `Strings` since they are unique and can not produce bugs because we just pass in Strings. 

If a invalid type is added a TypeError will be thrown, which should give enough information.

About the `DEL` type, this type is currently just in there for testing purposes it might be that this will be removed/replaced at a later time.

### CODE REVIEW

File: src/js/structs/Transaction.js [hunk 1]
```diff
@@ -1,5 +1,12 @@
+import FormActionTypes from '../constants/TransactionTypes';
```
Comment (MatApple) [inline src/js/structs/Transaction.js]: Maybe we should make this variable `TransactionTypes` just to keep things consistent.

File: src/js/structs/Transaction.js [hunk 2]
```diff
@@ -1,5 +1,12 @@
...
+
+const validTypes = Object.values(FormActionTypes);
+
 class Transaction {
-  constructor(path, value, type = 'SET') {
+  constructor(path, value, type = FormActionTypes.SET) {
+    if (!validTypes.includes(type)) {
+      throw new TypeError(`Only the following types are allowed: ${validTypes}`);
```
Comment (MatApple) [inline src/js/structs/Transaction.js]: The `validTypes` in the message will be Symbols. Perhaps we should output the `Object.keys(FormActionTypes)`.

File: src/js/structs/__tests__/Transaction-test.js [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
 describe('Transaction', function () {
   describe('#constructor', function () {
     it('should have the type SET', function () {
       const transaction = new Transaction(0, 0);
-      expect(transaction.type).toEqual('SET');
+      expect(transaction.type).toEqual(TransactionTypes.SET);
+    });
+
+    it('should throw a Error for random type', function () {
+      expect(()=>new Transaction(0, 0, 'DEL')).toThrowError(TypeError);
```
Comment (MatApple) [inline src/js/structs/__tests__/Transaction-test.js]: Lets space out the fat arrow notation `() => new ..`

File: src/js/structs/__tests__/Transaction-test.js [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
     });
 
-    it('should have the type DEL', function () {
-      const transaction = new Transaction(0, 0, 'DEL');
-      expect(transaction.type).toEqual('DEL');
+    it('should accept SET constant', function () {
+      expect(()=>new Transaction(0, 0, TransactionTypes.SET)).not.toThrowError();
```
Comment (MatApple) [inline src/js/structs/__tests__/Transaction-test.js]: and here

File: src/js/structs/Transaction.js (hunk 2 above)
Comment (Poltergeist) [inline src/js/structs/Transaction.js]: The error message right now is:
```
TypeError: Only the following types are allowed: Symbol(SET),Symbol(DEL)
```
I am okay with changing it to `SET, DEL, ...` , if that is still clear enough that only the predefined values are valid types and not strings.

File: src/js/structs/__tests__/Transaction-test.js (hunk 3 above)
Comment (Poltergeist) [inline src/js/structs/__tests__/Transaction-test.js]: done

File: src/js/structs/__tests__/Transaction-test.js (hunk 4 above)
Comment (Poltergeist) [inline src/js/structs/__tests__/Transaction-test.js]: done

**Pull request merged**